main.go

```diff
@@ -51,9 +51,15 @@ func run(ctx context.Context, credentials clients.Credentials, opts options.Opti
 
 	client := dalek.NewDalek(sdkClient, opts)
 	log.Printf("[DEBUG] Processing Resource Manager..")
-	if err := client.ResourceManager(ctx); err != nil {
-		return fmt.Errorf("processing Resource Manager: %+v", err)
+	if errors := client.ResourceManager(ctx); len(errors) != 0 {
+		errList := make([]string, 0)
+		for _, e := range errors {
+			errList = append(errList, "\n"+e.Error())
+		}
+
+		return fmt.Errorf("processing Resource Manager: %+v", errList)
```

Comment:
not a blocker, but it's typically easier to do the formatting during output, rather than in-line on each item fwiw:

```suggestion
		for _, e := range errors {
			errList = append(errList, e.Error())
		}

		return fmt.Errorf("processing Resource Manager: %+v", strings.Join(errList, "\n"))
```